bindings/wallet-cordova/scripts/build_jni.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import subprocess
+import sys
+import shutil
+from copy_jni_definitions import run as copy_definitions
+from directories import rust_build_directory, plugin_directory
+
+libname = "libwallet_jni.so"
+android_libs_directory = Path("src/android/libs")
+
+targets = {
+    "aarch64-linux-android": "arm64-v8a",
+    "armv7-linux-androideabi": "armeabi-v7a",
+    "i686-linux-android": "x86",
+    "x86_64-linux-android": "x86_64",
+}
+
+
+def copy_libs(release=True):
+    for rust_target, android_target in targets.items():
+        dst = plugin_directory / android_libs_directory / android_target
+        dst.mkdir(parents=True, exist_ok=True)
+
+        debug_or_release = "release" if release else "debug"
+
+        src = rust_build_directory / rust_target / debug_or_release / libname
+        shutil.copy(src, dst)
+
+
+def run(release=True):
+    for rust_target, android_target in targets.items():
```

Comment:
Why do all targets need to be built at once?

This is why you "need" to split the build into `build_jni` jobs and this single Android job, and pass temporary artefacts around. Maybe put these target configurations into the matrix config of the unified Android build job that is run on a per-target basis:

```yaml
      matrix:
        config:
          - { target: aarch64-linux-android, android_arch: arm64-v8a }
          - { target: armv7-linux-androideabi, android_arch: arbeabi-v7a }
          # ...
```

, and pass `target` and `android_arch` as parameters for this build script?

Reply:
https://github.com/input-output-hk/chain-wallet-libs/blob/b81fd1efb2cdf1f6bb6044569ce554c847c11288/bindings/wallet-cordova/plugin.xml#L33

These instructions that we have are not conditional copies, so the plugin build fails if they are not in place.

The plugin is a single package, and then when the app that uses it gets built the correct binary is selected by the android (gradle?) process, so I found it simpler to always build all of them.

That said, for testing is somewhat inconvenient, as usually when developing you're not testing all of the architectures and it wastes a bit of time. Because of this it may be worth to see if we can make the thing build partially for development.

The reason I had to pass artifacts around though, instead of just using the `build_jni.py`, is that the script uses `cross` inside, and that doesn't work inside the docker container on the CI environment. But using the `use-cross` flag of the `cargo` action does work.